scripts/Generate_ScopedFragmentTransformer.linq

```diff
@@ -61,52 +63,101 @@ void Main() {
     sb.AppendLine("");
 
     var first = true;
-    foreach (var t in VisitableTypes().OrderBy(t => t.Name)) {
+    var visitableTypes = VisitableTypes().OrderBy(t => t.Type.Name).ToArray();
+    foreach (var t in visitableTypes.Where(t => t.HasExplicitVisitMethod).Select(t => t.Type)) {
         if (!first) {
             sb.AppendLine($"\n");
         }
         sb.AppendLine($"    public override void ExplicitVisit({t.Name} node) {{");
         sb.AppendLine($"        if (SkipList.Contains(node)) {{ return; }}");
         sb.AppendLine($"        if (ShouldStop) {{ return; }}");
+        if (t.BaseType is Type parentType
+            && parentType != typeof(object)) {
+            sb.AppendLine($"        if (VisitParentTypes) {{");
+            sb.AppendLine($"            this.ExplicitBaseVisit(({parentType.Name})node);");
+            sb.AppendLine($"            if (ShouldStop) {{ return; }}");
+            sb.AppendLine($"        }}");
+        }
         sb.AppendLine($"        {VisitFnName(t)}?.Invoke(this, node);");
         sb.AppendLine($"        if (ShouldStop) {{ return; }}");
         sb.AppendLine($"");
         sb.AppendLine($"        PushContext(node);");
         sb.AppendLine($"        base.ExplicitVisit(node);");
+
         sb.AppendLine($"        PopContext();");
         sb.AppendLine($"");
         sb.AppendLine($"        HandleOnLeave(node);");
+
+
+
+        sb.AppendLine($"    }}");
+        first = false;
+    }
+
+    foreach (var t in visitableTypes.Where(t => !t.HasExplicitVisitMethod).Select(t => t.Type)) {
+        sb.AppendLine($"\n");
+        sb.AppendLine($"    public void ExplicitBaseVisit({t.Name} node) {{");
+        sb.AppendLine($"        if (ShouldStop) {{ return; }}");
+        if (t.BaseType is Type parentType
+            && parentType != typeof(object)) {
+            sb.AppendLine($"        if (VisitParentTypes) {{ this.ExplicitBaseVisit(({parentType.Name})node); }}");
+        }

```

Comment:
Should these two be flipped so that we check for stoppage after visiting the base?

Reply:
Wait, or before and after visiting the base? With my suggestion, it'll call all the way up the tree even if ShouldStop is true. With as it is currently written, it'll Invoke the visitors on the way back down the tree if ShouldStop is true.

Reply:
This is an extra place we could call it. Is that what you mean?

![image](https://github.com/user-attachments/assets/0de193a8-f10f-40b9-82f8-c5cbe36c17d4)

Reply:
Yes. My first thought was moving it to there, but my second thought was adding it there.

Reply:
If it's not necessary for the flow of what you're doing for some other reason, that's fine with me too.

Reply:
Ok. For me it isn't necessary, and each one of these adds a ton of code that is making it a pain to work with.

Reply:
Then this is good.